src/azure-cli/azure/cli/command_modules/synapse/_params.py

```diff
@@ -251,29 +251,44 @@ def load_arguments(self, _):
-                   arg_group=policy_arg_group,
-                   help='Auditing policy state',
-                   arg_type=get_enum_type(BlobAuditingPolicyState))
-        c.argument('audit_actions_and_groups',
-                   options_list=['--actions'],
-                   arg_group=policy_arg_group,
-                   help='List of actions and action groups to audit.',
-                   nargs='+')
-        c.argument('retention_days',
-                   arg_group=policy_arg_group,
-                   help='The number of days to retain audit logs.')
+    for scope in ['synapse sql pool audit-policy', 'synapse sql audit-policy']:
+        with self.argument_context(scope + ' update') as c:
+            _configure_security_or_audit_policy_storage_params(c)
+            c.argument('storage_account_subscription_id', arg_group=storage_arg_group,
+                       options_list=['--storage-subscription'],
+                       help='The subscription id of storage account')
+            c.argument('is_storage_secondary_key_in_use', arg_group=storage_arg_group,
+                       arg_type=get_three_state_flag(), options_list=['--use-secondary-key'],
+                       help='Indicates whether using the secondary storeage key or not')
+            c.argument('is_azure_monitor_target_enabled', options_list=['--enable-azure-monitor'],
+                       help='Whether enabling azure monitor target or not.',
+                       arg_type=get_three_state_flag())
+            c.argument('state',
+                       arg_group=policy_arg_group,
+                       help='Auditing policy state',
+                       arg_type=get_enum_type(BlobAuditingPolicyState))
+            c.argument('audit_actions_and_groups',
+                       options_list=['--actions'],
+                       arg_group=policy_arg_group,
+                       help='List of actions and action groups to audit.',
+                       nargs='+')
+            c.argument('retention_days',
+                       arg_group=policy_arg_group,
+                       help='The number of days to retain audit logs.')
+
+    with self.argument_context('synapse sql audit-policy update') as c:
+        c.argument('queue_delay_milliseconds', type=int,
+                   options_list=['--queue-delay-time', '--queue-delay-milliseconds'],
+                   help='The amount of time in milliseconds that can elapse before audit actions are forced to be processed')
+
+    with self.argument_context('synapse sql ad-admin') as c:
+        c.argument('workspace_name', help='The workspace name.')
+    for scope in ['create', 'update']:
+        with self.argument_context('synapse sql ad-admin ' + scope) as c:
+            c.argument('login_name', options_list=['--display-name', '-u'],
+                       help='Display name of the Azure AD administrator user or group.')
+            c.argument('object_id', options_list=['--object-id', '-i'],
```

Comment:
question: would these two parameters are both required for the commands?
I am wondering why object id could not determine the admin and we need name here.

Reply:
> question: would these two parameters are both required for the commands?
> I am wondering why object id could not determine the admin and we need name here.

Per my test, object_id is mandatory, the display_name is not and the display_name can be any string. My understanding iis display name is more friendly and  we do not have the logic that get display name by object id **in cli**.

Reply:
interesting. In this way, what is the role for display name in your funtion?

Reply:
> interesting. In this way, what is the role for display name in your funtion?

The display name provided by customer will be shown in portal.

Reply:
The display name should be bound with object id and it is pre-defined for the admin, right?
From my previous understanding, the display name is already fixed with object id.
Could your command redefine the display name?

Reply:
We need to keep consistent with `az sql server ad-admin`.
In synapse `az synapse sql ad-admin create` the display name and object id are both mandatory. And the display name is no fixed, it can be anything, it just display in portal.

![image](https://user-images.githubusercontent.com/12733681/102857847-2306d200-4464-11eb-8d23-915dfb258c33.png)